addons/l10n_id_efaktur/models/account_move.py

```diff
@@ -0,0 +1,314 @@
+# -*- encoding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import base64
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError, ValidationError
+
+fk_head_list = ['FK', 'KD_JENIS_TRANSAKSI', 'FG_PENGGANTI', 'NOMOR_FAKTUR', 'MASA_PAJAK', 'TAHUN_PAJAK', 'TANGGAL_FAKTUR', 'NPWP', 'NAMA', 'ALAMAT_LENGKAP', 'JUMLAH_DPP', 'JUMLAH_PPN', 'JUMLAH_PPNBM', 'ID_KETERANGAN_TAMBAHAN', 'FG_UANG_MUKA', 'UANG_MUKA_DPP', 'UANG_MUKA_PPN', 'UANG_MUKA_PPNBM', 'REFERENSI']
+
+lt_head_list = ['LT', 'NPWP', 'NAMA', 'JALAN', 'BLOK', 'NOMOR', 'RT', 'RW', 'KECAMATAN', 'KELURAHAN', 'KABUPATEN', 'PROPINSI', 'KODE_POS', 'NOMOR_TELEPON']
+
+of_head_list = ['OF', 'KODE_OBJEK', 'NAMA', 'HARGA_SATUAN', 'JUMLAH_BARANG', 'HARGA_TOTAL', 'DISKON', 'DPP', 'PPN', 'TARIF_PPNBM', 'PPNBM']
+
+
+def _csv_row(data, delimiter=',', quote='"'):
+    return quote + (quote + delimiter + quote).join([str(x).replace(quote, '\\' + quote) for x in data]) + quote + '\n'
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    l10n_id_efaktur_id = fields.Many2one('l10n_id_efaktur.efaktur', string="E-faktur", copy=False)
+    l10n_id_tax_number = fields.Char(string="Tax Number", inverse='_inverse_tax_number', compute='_compute_tax_number', store=False, copy=False)
+    l10n_id_replace_invoice_id = fields.Many2one('account.move', string="Replace Invoice",  domain="['|', '&', '&', ('state', '=', 'posted'), ('partner_id', '=', partner_id), ('reversal_move_id', '!=', False), ('state', '=', 'cancel')]", copy=False)
+    l10n_id_upload_user = fields.Many2one('res.users', 'Upload User', readonly=True, copy=False)
+    l10n_id_attachment_id = fields.Many2one('ir.attachment', readonly=True, copy=False)
```

Comment:
can we use a better name than attachment_id?  What is it?

Reply:
It is the csv like file to upload to the tax office website, do you have a proposition?